src/types/render-state.ts

```diff
@@ -1,223 +1,68 @@
-  HierarchicalMenuConnectorParams,
-  HierarchicalMenuRendererOptions,
-} from '../connectors/hierarchical-menu/connectHierarchicalMenu';
-import {
-  RefinementListRendererOptions,
-  RefinementListConnectorParams,
-} from '../connectors/refinement-list/connectRefinementList';
-import {
-  StatsConnectorParams,
-  StatsRendererOptions,
-} from '../connectors/stats/connectStats';
-import {
-  SortByConnectorParams,
-  SortByRendererOptions,
-} from '../connectors/sort-by/connectSortBy';
+import { AnswersWidgetDescription } from '../connectors/answers/connectAnswers';
+import { AutocompleteWidgetDescription } from '../connectors/autocomplete/connectAutocomplete';
+import { BreadcrumbWidgetDescription } from '../connectors/breadcrumb/connectBreadcrumb';
+import { ClearRefinementsWidgetDescription } from '../connectors/clear-refinements/connectClearRefinements';
+import { ConfigureWidgetDescription } from '../connectors/configure/connectConfigure';
+import { CurrentRefinementsWidgetDescription } from '../connectors/current-refinements/connectCurrentRefinements';
+import { GeoSearchWidgetDescription } from '../connectors/geo-search/types';
+import { HierarchicalMenuWidgetDescription } from '../connectors/hierarchical-menu/connectHierarchicalMenu';
+import { HitsPerPageWidgetDescription } from '../connectors/hits-per-page/connectHitsPerPage';
+import { HitsWidgetDescription } from '../connectors/hits/connectHits';
+import { InfiniteHitsWidgetDescription } from '../connectors/infinite-hits/connectInfiniteHits';
+import { MenuWidgetDescription } from '../connectors/menu/connectMenu';
+import { NumericMenuWidgetDescription } from '../connectors/numeric-menu/connectNumericMenu';
+import { PaginationWidgetDescription } from '../connectors/pagination/connectPagination';
+import { PoweredByWidgetDescription } from '../connectors/powered-by/connectPoweredBy';
+import { QueryRulesWidgetDescription } from '../connectors/query-rules/connectQueryRules';
+import { RangeWidgetDescription } from '../connectors/range/connectRange';
+import { RatingMenuWidgetDescription } from '../connectors/rating-menu/connectRatingMenu';
+import { RefinementListWidgetDescription } from '../connectors/refinement-list/connectRefinementList';
+import { RelevantSortWidgetDescription } from '../connectors/relevant-sort/connectRelevantSort';
+import { SearchBoxWidgetDescription } from '../connectors/search-box/connectSearchBox';
+import { SortByWidgetDescription } from '../connectors/sort-by/connectSortBy';
+import { StatsWidgetDescription } from '../connectors/stats/connectStats';
+import { ToggleRefinementWidgetDescription } from '../connectors/toggle-refinement/types';
+import { VoiceSearchWidgetDescription } from '../connectors/voice-search/connectVoiceSearch';
+import { AnalyticsWidgetDescription } from '../widgets/analytics/analytics';
+import { PlacesWidgetDescription } from '../widgets/places/places';
+
+type ConnectorRenderStates = AnswersWidgetDescription['indexRenderState'] &
```

Comment:
shouldn't we use the Expand helper here?

Reply:
there's no conditionals, so intellisense automatically expanded the type already :)